pandas/core/groupby.py

```diff
@@ -1021,6 +1022,45 @@ def cumcount(self, ascending=True):
         cumcounts = self._cumcount_array(ascending=ascending)
         return Series(cumcounts, index)
 
+    def cumprod(self, axis=0):
+        """
+        Cumulative product for each group
+
+        """
+        if axis != 0:
```

Comment:
The transforms operate by block, so that doesn't work directly.  Although if we're grouping along axis=0,  I suppose could call out to `obj.cumprod(axis=1)` since the grouping doesn't actually do anything in that case.